test/e2e/network/ingress.go

```diff
@@ -1112,19 +1112,40 @@ var _ = SIGDescribe("Ingress API", func() {
 
 		// Ingress resource delete operations
 		ginkgo.By("deleting")
+
+		expectFinalizer := func(ing *networkingv1.Ingress) {
+			framework.ExpectNotEqual(ing.DeletionTimestamp, nil, fmt.Sprintf("expected deletionTimestamp, got nil, ingress: %+v", ing))
+			framework.ExpectEqual(len(ing.Finalizers) > 0, true, fmt.Sprintf("expected finalizers on ingress, got none, ingress: %+v", ing))
+		}
+
 		err = ingClient.Delete(context.TODO(), createdIngress.Name, metav1.DeleteOptions{})
 		framework.ExpectNoError(err)
-		_, err = ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
-		framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %#v", err))
+		ing, err := ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
+		// If ingress controller does not support finalizers, we expect a 404.  Otherwise we validate finalizer behavior.
+		if err == nil {
+			expectFinalizer(ing)
+		} else {
+			framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %v", err))
+		}
 		ings, err = ingClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(len(ings.Items), 2, "filtered list should have 2 items")
+		// Should have <= 3 items since some ingresses might not have been deleted yet due to finalizers
+		framework.ExpectEqual(len(ings.Items) <= 3, true, "filtered list should have <= 3 items")
+		// Validate finalizers
+		for _, ing := range ings.Items {
+			expectFinalizer(&ing)
```

Comment:
you only deleted `createdIngress.Name`, you should only expect a finalizer on that one:

```suggestion
for _, ing := range ings.Items {
	if ing.Namespace == createdIngress.Namespace && ing.Name == createdIngress.Name {
		// the ingress we deleted should only still exist if it has a finalizer
		expectFinalizer(&ing)
	}
}
```

Reply:
Ah good catch 🤦‍♂️